### PR TITLE
Reimplement appbar color overlay on scroll

### PR DESCRIPTION
**Pull request by ivaniskandar**
only on updates and history screen for now, but the required changes on app bar
is there.

also fix missing incognito-downloaded mode indicator on history screen

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt [hunk 1]
```diff
@@ -45,11 +46,16 @@ class HistoryPresenter(
     private val deleteHistoryTable: DeleteHistoryTable = Injekt.get(),
     private val removeHistoryById: RemoveHistoryById = Injekt.get(),
     private val removeHistoryByMangaId: RemoveHistoryByMangaId = Injekt.get(),
+    preferences: PreferencesHelper = Injekt.get(),
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:49]: Can be private?

File: app/src/main/java/eu/kanade/presentation/components/AppBar.kt [hunk 2]
```diff
@@ -51,19 +48,63 @@ fun AppBar(
...
+        incognitoMode = incognitoMode,
+        scrollBehavior = scrollBehavior,
+    )
+}
+
+@Composable
+fun AppBar(
+    modifier: Modifier = Modifier,
+    // Title
+    titleContent: @Composable () -> Unit,
```
Comment (arkon) [inline app/src/main/java/eu/kanade/presentation/components/AppBar.kt:85]: Is this in preparation for different content or something?

File: app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt (hunk 1 above)
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:49]: oopsie

File: app/src/main/java/eu/kanade/presentation/components/AppBar.kt (hunk 2 above)
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/presentation/components/AppBar.kt:85]: It's currently used for history search

File: app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt (hunk 1 above)
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:49]: wait no this is just a parameter not a property

Comment (arkon) [reply app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:49]: Whoops, you right

**Pull request merged**